src/modules/Generator.jl

```diff
@@ -0,0 +1,86 @@
+"""
+Provides the functions related to generating documentation stubs.
+"""
+module Generator
+
+"""
+    savefile(f, root, filename)
+
+Attempts to save a file at `\$(root)/\$(filename)`. `f` will be called with file
+stream (see [`open`](http://docs.julialang.org/en/latest/stdlib/io-network/#Base.open)).
+"""
+function savefile(f, root, filename)
+    filepath = joinpath(root, filename)
+    if ispath(filepath) error("$(filepath) already exists") end
+    open(f,filepath,"w")
+end
+
+"""
+Contents of the default `make.jl` file.
+"""
+function make(pkgname)
+    """
+    using Documenter
+    using $(pkgname)
+
+    makedocs(
+        modules = [$(pkgname)]
+    )
+
+    #deploydocs()
+    """
+end
+
+"""
+Contents of the default `.gitignore` file.
+"""
+function gitignore()
+    """
+    build/
+    site/
+    """
+end
+
+"""
+Contents of the default `mkdocs.yml` file.
+"""
+function mkdocs(pkgname)
+    """
+    site_name:        $(pkgname).jl
+    #repo_url:         https://github.com/USER_NAME/PACKAGE_NAME.jl
+    #site_description: Description...
+    #site_author:      USER_NAME
```

Comment:
See [`getremote`](https://github.com/MichaelHatherly/Documenter.jl/blob/3b76ac806c42a5580354a0c4d42ac31d8cfc020a/src/modules/Utilities.jl#L355-L369) for how to get at the `origin` in `0.5`, it's also possible to get in `0.4`, but since `Base.LibGit2` doesn't exist there you need to shell out to `git` directly.

For the moment we can just got with the placeholder text since users will need to make some changes to the file any way.